Sources/Rendering/OpenGL/ShaderProgram/index.js

```diff
@@ -184,6 +184,26 @@ export function vtkShaderProgram(publicAPI, model) {
     return true;
   };
 
+  publicAPI.setUniform4f = (name, v) => {
+    const location = publicAPI.findUniform(name);
+    if (location === -1) {
+      model.error = `Could not set uniform ${name} . No such uniform.`;
+      return false;
+    }
+    model.context.uniform4f(location, v[0], v[1], v[2], v[3]);
+    return true;
+  };
+
+  publicAPI.setUniform4fv = (name, count, v) => {
+    const location = publicAPI.findUniform(name);
+    if (location === -1) {
+      model.error = `Could not set uniform ${name} . No such uniform.`;
+      return false;
+    }
+    model.context.uniform4fv(location, count, v);
```

Comment:
@jourdain  @jpouderoux 
I tried to use this function and it is not working because uniform4fv method needs 2 parameters instead of 3.
[](https://developer.mozilla.org/en-US/docs/Web/API/WebGLRenderingContext/uniform)

I tried to put only `model.context.uniform4fv(location, v);` but it returns the following error:" Expected an array of length a multiple of 4, got an array of length 1."

A solution here is the following one:
```
for (let i = 0; i<v.length; i++) {
        model.context.uniform4fv(location, v[i]);
}
```
I tested it, and it works, I am not sure about the quality of it.